doc/west/flash-debug.rst

```diff
@@ -0,0 +1,243 @@
+.. _west-flash-debug:
+
+West: Flashing and Debugging
+############################
+
+West provides three commands for running and interacting with Zephyr
+programs running on a board: ``flash``, ``debug``, and
+``debugserver``.
+
+These use information stored in the CMake cache to flash or attach a
```

Comment:
More of a comment, but as a CMake noob, I'd appreciate a short description of what the CMake cache is, or a link to the CMake documentation.